shared/AppInsightsCommon/src/StorageHelperFuncs.ts

```diff
@@ -88,142 +165,108 @@ export function utlEnableStorage() {
-        }
+        storage.setItem(name, data);
+        return true;
     }
     return false;
 }
 
 export function utlRemoveStorage(logger: IDiagnosticLogger, name: string): boolean {
     const storage = _getLocalStorageObject();
     if (storage !== null) {
-        try {
-            storage.removeItem(name);
-            return true;
-        } catch (e) {
-            _canUseLocalStorage = false;
-
-            _throwInternal(logger,
-                eLoggingSeverity.WARNING,
-                _eInternalMessageId.BrowserFailedRemovalFromLocalStorage,
-                "Browser failed removal of local storage item. " + getExceptionName(e),
-                { exception: dumpObj(e) });
-        }
+        storage.removeItem(name);
+        return true;
     }
     return false;
 }
 
 export function utlCanUseSessionStorage(reset?: boolean): boolean {
-    if (reset || _canUseSessionStorage === undefined) {
-        _canUseSessionStorage = !!_getVerifiedStorageObject(StorageType.SessionStorage);
+    if (reset) {
+        _verifiedSessionStorage = null;
     }
-
-    return _canUseSessionStorage;
+    
+    const storage = _getSessionStorageObject();
+    return !!storage;
 }
 
 export function utlGetSessionStorageKeys(): string[] {
-    const keys: string[] = [];
-
     if (utlCanUseSessionStorage()) {
-        objForEachKey(getGlobalInst<any>("sessionStorage"), (key) => {
-            keys.push(key);
-        });
+        const keys: string[] = [];
+        try {
+            objForEachKey(getGlobalInst<any>("sessionStorage"), (key) => {
+                keys.push(key);
+            });
+        } catch (e) {
+            // Invalidate session storage on any error
+            _verifiedSessionStorage = null;
+        }
+        return keys;
     }
-
-    return keys;
+    return [];
 }
 
 export function utlGetSessionStorage(logger: IDiagnosticLogger, name: string): string {
-    const storage = _getSessionStorageObject();
-    if (storage !== null) {
-        try {
+    try {
+        const storage = _getSessionStorageObject();
+        if (storage !== null) {
             return storage.getItem(name);
-        } catch (e) {
-            _canUseSessionStorage = false;
-
-            _throwInternal(logger,
-                eLoggingSeverity.WARNING,
-                _eInternalMessageId.BrowserCannotReadSessionStorage,
-                "Browser failed read of session storage. " + getExceptionName(e),
-                { exception: dumpObj(e) });
         }
+    } catch (e) {
+        _throwInternal(logger, eLoggingSeverity.WARNING,
+            _eInternalMessageId.FailedToGetSessionStorageItem,
+            "Failed to get session storage: " + getExceptionName(e),
+            { exception: dumpObj(e) });
     }
     return null;
 }
 
 export function utlSetSessionStorage(logger: IDiagnosticLogger, name: string, data: string): boolean {
-    const storage = _getSessionStorageObject();
-    if (storage !== null) {
-        try {
+    try {
```

Comment:
revert to the original, keeping the _throwInternal, but no need to set the _canUseLocalStorage as this is now handled in the wrapped implementation.